io-classes/src/Control/Monad/Class/MonadTimer.hs

```diff
@@ -53,6 +53,8 @@ class Monad m => MonadDelay m where
   threadDelay d   = void . atomically . awaitTimeout =<< newTimeout d
 
 class (MonadSTM m, MonadDelay m) => MonadTimer m where
+  -- | Associated data type for 'timeout' API
+  --
```

Comment:
This doesn't really say anything that the syntax doesn't already say. Try and phrase it in terms of what it is for. Also note that it's not used with `timeout`, it's used with the "fancy" timeout API. Perhaps better to name them explicitly, e.g.

> -- | The type of the timeout handle, used with 'newTimeout', 'readTimeout', 'updateTimeout' and 'cancelTimeout'.

Reply:
Done